python/pfs/drp/stella/coaddSpectra.py

```diff
@@ -33,6 +33,21 @@
 __all__ = ("CoaddSpectraConfig", "CoaddSpectraTask")
 
 
+class SetWithNaN(set):
+    """A set which includes at most one NaN, despite the fact that NaN != NaN"""
+    def __init__(self, iterable):
+        """Construct a set from iterable with at most one NaN"""
+        sawNaN = False
+
+        for x in iterable:
+            if np.isnan(x):
+                if not sawNaN:
+                    self.add(x)
+                    sawNaN = True
+            else:
+                self.add(x)
```

Comment:
I expect this does what we need, but if we were to use this more widely it would need a check in the `add` method as well.